src/util/charbuf.c

```diff
@@ -13,20 +13,38 @@ charbuf new_charbuf(size_t len)
 {
   charbuf newBuf;
 
-  newBuf.chars = (unsigned char *) malloc(len);
-  newBuf.len = len;
+  newBuf.len = 0;
```

Comment:
I think we can clean this a little by setting newBuf.chars to NULL first.

    charbuf new_charbuf(size_t len)
    {
      charbuf newBuf;

      newBuf.len = 0;
      newBuf.chars = NULL;
  
      if (len > 0)
      { 
        newBuf.chars = (unsigned char *) malloc(len);
        if (newBuf.chars != NULL)
        { 
          newBuf.len = len;
        }
      }
      return (newBuf);
    }